odd-collector/odd_collector/adapters/postgresql/mappers/relationships/identifying_checker.py

```diff
@@ -0,0 +1,33 @@
+from odd_models.models.models import DataSetField
+
+
+class IdentifyingChecker:
+    def __init__(self,
+                 foreign_key: tuple[str],
+                 ref_foreign_key: tuple[str],
+                 source_field_list: list[DataSetField],
+                 target_field_list: list[DataSetField]):
+        self.foreign_key = foreign_key
+        self.ref_foreign_key = ref_foreign_key
+        self.source_field_list = source_field_list
+        self.target_field_list = target_field_list
+
+    def is_identifying(self) -> bool:
+        return self._is_fk_part_of_pk() and self._is_ref_fk_refers_to_pk()
+
+    def _is_fk_part_of_pk(self) -> bool:
+        """
+        Check if the foreign key is part of primary key source table
+        """
+        fk_columns = {dsf.name for dsf in self.source_field_list if dsf.name in self.foreign_key}
```

Comment:
I think it is overhead for getting set of `fk_columns`. We already have the tuple of them in `self.foreign_key` so set creation can be simply: `fk_columns  = set(self.foreign_key)` without redundant checks over the whole `self.source_field_list`.